### ISSUE TITLE
Add envrc-mode-hook

### ISSUE THREAD

**Issue by manuel-uberti** [purcell/envrc#1]
Hi,

first of all thanks for this package, it addresses exactly what I think was missing from `direnv.el`.

Do you think it would be possible to add a `envrc-mode-hook` to set up other per-user stuff after the mode is enabled? More importantly, do you think it would be something useful?

I have a font-lock related function for `.envrc` files that I'd like to run after `envrc-global-mode`. Of course I can wrap the call to `envrc-global-mode` in a `defun` where I call my font-lock utility as well, and use that.

So, basically, it's your call. :)

**Comment by purcell** [comment]
There will already be an `envrc-mode-hook` for each buffer and an `envrc-global-mode-hook` globally. Does that give you what you need?

**Comment by manuel-uberti** [comment]
Absolutely, thank you.

**Comment by purcell** [comment]
I can imagine it being helpful to have a global and/or local hook that is triggered when the state changes for a direnv, for example: the existing hook doesn't really give you that.

**Comment by purcell** [comment]
BTW, in reading your issue description, I couldn't really tell how the font-lock functionality would be related: this minor mode isn't really related to editing those files. (`direnv.el` does have a simple major mode for `.envrc` files, and I might add something similar, perhaps.)

**Comment by manuel-uberti** [comment]
Yes, my font-locking trick isn't actually working at the moment with your package, but I can live without it because of what this package addresses.

If you think a mode for `.envrc` files can be useful, I'll be using it, but it's not a blocking issue at all.
